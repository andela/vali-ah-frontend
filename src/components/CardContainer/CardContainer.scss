.card-container {
  display: flex;
  justify-content: space-between;
  width: 80%;
  margin: 0 auto;
  text-align: left;
  flex-wrap: wrap;

  &__amazing-topics {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-bottom: 2rem;
  }
}

@media (max-width: 48rem) {
  .card-container {
    justify-content: center;
    width: 100%;
  }
}
