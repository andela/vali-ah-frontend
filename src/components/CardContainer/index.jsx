/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';

import Proptypes from 'prop-types';

import Masonry from 'components/Masonry';

import './CardContainer.scss';

const CardContainer = (props) => {
  const { CardComponent, cardData } = props;
  return (
    <div className="app-container card-container">
      <div className="card-container card-container__amazing-topics">
        <h1>Explore Amazing Topics</h1>
        <p>Curated stories based on your preference</p>
      </div>
      <Masonry>
        {(cardData || []).map((data) => <CardComponent {...data} />)}
      </Masonry>
    </div>
  );
};

CardContainer.propTypes = {
  cardData: Proptypes.arrayOf().isRequired,
  CardComponent: Proptypes.node.isRequired
};

export default CardContainer;
