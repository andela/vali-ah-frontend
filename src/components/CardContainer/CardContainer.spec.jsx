import React from 'react';

import CardContainer from './index';


describe('CardContainer Component', () => {
  it('should renders properly without card data', () => {
    const cardContainer = mount(<CardContainer />);

    expect(cardContainer).toMatchSnapshot();
  });

  it('should renders properly with card data', () => {
    const data = [{
      title: '',
      summary: '',
      category: ''
    }];
    const CardComponent = () => () => <span>MultiSelect</span>;
    const cardContainer = mount(<CardContainer cardData={data} CardComponent={CardComponent} />);

    expect(cardContainer).toMatchSnapshot();
  });
});
