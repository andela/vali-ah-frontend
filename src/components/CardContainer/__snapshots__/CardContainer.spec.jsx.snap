// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Component should renders properly with card data 1`] = `
<CardContainer
  CardComponent={[Function]}
  cardData={
    Array [
      Object {
        "category": "",
        "summary": "",
        "title": "",
      },
    ]
  }
>
  <div
    className="app-container card-container"
  >
    <div
      className="card-container card-container__amazing-topics"
    >
      <h1>
        Explore Amazing Topics
      </h1>
      <p>
        Curated stories based on your preference
      </p>
    </div>
    <Masonry
      gap=""
      minWidth={300}
    >
      <styled.div
        gap=""
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "keyKcu",
                "rules": Array [
                  "
  display: grid;
  grid-auto-flow: column;
  grid-gap: ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={
            Object {
              "current": <div
                class="sc-bdVaJa keyKcu"
              >
                <div
                  class="sc-bwzfXH itrbwG"
                />
                <div
                  class="sc-bwzfXH itrbwG"
                />
                <div
                  class="sc-bwzfXH itrbwG"
                />
              </div>,
            }
          }
          gap=""
        >
          <div
            className="sc-bdVaJa keyKcu"
          >
            <styled.div
              gap=""
              key="0"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
            <styled.div
              gap=""
              key="1"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
            <styled.div
              gap=""
              key="2"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Masonry>
  </div>
</CardContainer>
`;

exports[`CardContainer Component should renders properly without card data 1`] = `
<CardContainer>
  <div
    className="app-container card-container"
  >
    <div
      className="card-container card-container__amazing-topics"
    >
      <h1>
        Explore Amazing Topics
      </h1>
      <p>
        Curated stories based on your preference
      </p>
    </div>
    <Masonry
      gap=""
      minWidth={300}
    >
      <styled.div
        gap=""
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "keyKcu",
                "rules": Array [
                  "
  display: grid;
  grid-auto-flow: column;
  grid-gap: ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={
            Object {
              "current": <div
                class="sc-bdVaJa keyKcu"
              >
                <div
                  class="sc-bwzfXH itrbwG"
                />
                <div
                  class="sc-bwzfXH itrbwG"
                />
                <div
                  class="sc-bwzfXH itrbwG"
                />
              </div>,
            }
          }
          gap=""
        >
          <div
            className="sc-bdVaJa keyKcu"
          >
            <styled.div
              gap=""
              key="0"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
            <styled.div
              gap=""
              key="1"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
            <styled.div
              gap=""
              key="2"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "itrbwG",
                      "rules": Array [
                        "
  display: grid;
  grid-gap: ",
                        [Function],
                        ";
  grid-auto-rows: unset;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                gap=""
              >
                <div
                  className="sc-bwzfXH itrbwG"
                />
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </Masonry>
  </div>
</CardContainer>
`;
